services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/IntersectionController.java

```diff
@@ -0,0 +1,68 @@
+package us.dot.its.jpo.ode.api.controllers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import us.dot.its.jpo.ode.api.accessors.map.ProcessedMapRepository;
+import us.dot.its.jpo.ode.api.models.IntersectionReferenceData;
+
+@RestController
+@ConditionalOnProperty(name = "enable.api", havingValue = "true", matchIfMissing = false)
+public class IntersectionController {
+
+    @Autowired
+    ProcessedMapRepository processedMapRepo;
+
+    @RequestMapping(value = "/intersection/list", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || @PermissionService.hasRole('USER')")
+    public ResponseEntity<List<IntersectionReferenceData>> getIntersections(
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            IntersectionReferenceData ref = new IntersectionReferenceData();
+            ref.setRsuIP("10.11.81.12");
```

Comment:
question: this isn't any ip/intersectionid that we actually use is it? Just double checking we aren't exposing data

Reply:
I updated all testing/mock IPs to 1.1.1.1 to ensure nothing is leaked